networkx/algorithms/bipartite/tests/test_edgecoloring.py

```diff
@@ -0,0 +1,302 @@
+"""Unit tests for the :mod:`networkx.algorithms.bipartite.edge_coloring` module."""
+from collections import defaultdict
+
+import pytest
+
+import networkx as nx
+from networkx.algorithms.bipartite.edge_colorings import edge_coloring
+
+
+def _is_proper_edge_coloring(G, coloring):
+    """Checks through each node and saves the colors at each node to find out
+    if there is any conflict
+    """
+    node_colors = defaultdict(set)
+
+    # iterate through each edge in the graph
+
+    if G.is_multigraph():
+        edges = G.edges(keys=True)
+    else:
+        edges = G.edges()
+
+    for edge in edges:
+        if isinstance(G, nx.MultiGraph):
+            u, v, key = edge
+            e1 = (u, v, key)
+            e2 = (v, u, key)
+        else:
+            u, v = edge
+            e1 = (u, v)
+            e2 = (v, u)
+
+        if e1 not in coloring:
+            return False
+        if e2 not in coloring:
+            return False
+        if coloring[e1] != coloring[e2]:
+            return False
+        else:
+            color = coloring[e1]
```

Comment:
It would be nice to have the `if` clauses turned into `assert` statements.
That's so when they fail we can tell which clause is failing.